OpenRA.Mods.Common/Traits/Buildings/BaseProvider.cs

```diff
@@ -80,7 +80,7 @@ bool ValidRenderPlayer()
 		public IEnumerable<IRenderable> RangeCircleRenderables(WorldRenderer wr)
 		{
 			// Visible to player and allies
-			if (!ValidRenderPlayer())
+			if (!ValidRenderPlayer() || !wr.World.WorldActor.Trait<MapBuildRadius>().BuildRadiusEnabled)
```

Comment:
I think we should cache `BuildRadiusEnabled` here as well as `AllyBuildEnabled` is already. (Also see the comment below.)